src/components/filter/DropdownFilter.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from 'react';
+import { Controller } from 'react-hook-form';
+import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
+
+import { DropdownSelect } from '../DropdownSelect';
+import { Label } from '../Label';
+
+import { Input } from './../form';
+import { normalize } from '../../config';
+
+type Props = {
+  containerStyle?: StyleProp<ViewStyle>;
+  control: any;
+  data: {
+    selected: boolean;
+    value: string;
+  }[];
+  displayKey?: string;
+  errors: any;
+  label?: string;
+  name: string;
+  placeholder?: string;
+  required?: boolean;
+  valueKey: string;
+};
+
+export const DropdownFilter = ({
+  containerStyle,
+  control,
+  data,
+  displayKey = 'value',
+  errors,
+  label,
+  name,
+  placeholder,
+  required,
+  valueKey = 'value'
+}: Props) => {
+  const dropdownEntries = [{ selected: true, [displayKey]: placeholder }, ...data];
+
+  const [dropdownData, setDropdownData] = useState(
+    dropdownEntries.map((item) => ({ ...item, value: item[displayKey] }))
+  );
+
+  return (
+    <>
+      <Label style={{ marginBottom: normalize(-10) }}>{label}</Label>
+      <View style={(styles.container, containerStyle)}>
+        <Controller
+          name={name}
+          render={({ field: { name, onChange } }) => {
+            useEffect(() => {
+              onChange(
```

Comment:
It may be enough, but I used `Controller` to capture the filtered data in the `onFilter` function in the `Filter` component.

Reply:
I removed `react-hook-form` as you said because I think it improves performance and I linked all data with `setFilters` state. 

https://github.com/smart-village-solutions/smart-village-app-app/pull/1017/commits/092dad9d4b8a73955f2dd028d900414047b1c9a1